### PR TITLE
Enumo scheduler

### PR DESCRIPTION
**Pull request by ajpal**
Make it easy to swap in different scheduling strategies in enumo recipes

### CODE REVIEW

File: src/enumo/scheduler.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+                for _ in 0..limits.iter {
+                    // Sat
+                    runner = Self::mk_runner(runner.egraph, &Limits::max()).run(&sat);
+
+                    // Other
+                    runner = Self::mk_runner(
+                        runner.egraph,
+                        &Limits {
+                            iter: 1,
```
Comment (chandrakananandi) [inline src/enumo/scheduler.rs:58]: oh we only run the other rules for a single iteration? should this be a param? Or will it always be 1?

Comment (chandrakananandi) [reply src/enumo/scheduler.rs:58]: ah I guess it will always be 1

File: src/enumo/scheduler.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                            node: limits.node,
```
Comment (chandrakananandi) [inline src/enumo/scheduler.rs:59]: should we have this limit here? what if the node limit is hit even before iter 1?

**Review by chandrakananandi — APPROVED** [review]

File: src/enumo/scheduler.rs (hunk 2 above)
Comment (ajpal) [reply src/enumo/scheduler.rs:59]: Yeah, I think it makes sense to have- otherwise we never use the node limit

**Pull request merged**